Develop

samuelouteda opened the pull request:
Assembled SideBar component:
![image](https://user-images.githubusercontent.com/74317636/100380270-ebd00d00-2fec-11eb-9cc2-c12b60823256.png)

ProfilePic component:
![image](https://user-images.githubusercontent.com/74317636/100380315-fe4a4680-2fec-11eb-8a38-5ff10b89dbb3.png)

NavLinks component:
![image](https://user-images.githubusercontent.com/74317636/100380389-20dc5f80-2fed-11eb-8001-e70ff87da544.png)

Footer component:
![image](https://user-images.githubusercontent.com/74317636/100380437-3ce00100-2fed-11eb-94bb-bff1a44fdedd.png)

Commit:
Creation of routes

Commit:
Creation of routes

samuelouteda committed:
Merge branch 'main' into develop

samuelouteda committed:
Creation of ProfilePic

Creation of the first component of sidebar, the ProfilePic, which contains a profile picture and a name

samuelouteda committed:
Creation of ProfilePic

Creation of the first component of sidebar, the ProfilePic, which contains a profile picture and a name

samuelouteda committed:
parent 174d1b0a5d265e377cd8500471288ef9a70cdf64
author Samuel Outeda <samuelouteda@gmail.com> 1605795684 -0400
committer Samuel Outeda <samuelouteda@gmail.com> 1606247868 -0400

parent 174d1b0a5d265e377cd8500471288ef9a70cdf64
author Samuel Outeda <samuelouteda@gmail.com> 1605795684 -0400
committer Samuel Outeda <samuelouteda@gmail.com> 1606247345 -0400

parent 174d1b0a5d265e377cd8500471288ef9a70cdf64
author Samuel Outeda <samuelouteda@gmail.com> 1605795684 -0400
committer Samuel Outeda <samuelouteda@gmail.com> 1606247188 -0400

parent 174d1b0a5d265e377cd8500471288ef9a70cdf64
author Samuel Outeda <samuelouteda@gmail.com> 1605795684 -0400
committer Samuel Outeda <samuelouteda@gmail.com> 1606247037 -0400

parent 174d1b0a5d265e377cd8500471288ef9a70cdf64
author Samuel Outeda <samuelouteda@gmail.com> 1605795684 -0400
committer Samuel Outeda <samuelouteda@gmail.com> 1606246990 -0400

parent 174d1b0a5d265e377cd8500471288ef9a70cdf64
author Samuel Outeda <samuelouteda@gmail.com> 1605795684 -0400
committer Samuel Outeda <samuelouteda@gmail.com> 1606246066 -0400

Creation of ProfilePic

Creation of the first component of sidebar, the ProfilePic, which contains a profile picture and a name

Creation of NavLinks

Creation of the second component of sidebar, the NavLinks, which contains a list of navigation links

Creating file to import all sections

Creating file to import all sections and export where needed

Simplifying imports

Using exports from index.js to simplify imports

Improving code

Reducing the amount of imports, converting functions into ES6 arrow function aways, adding missing end of line

Creation of Footer

Creation of the third and last component of sidebar, the Footer, which contains links to social networks with their icons and copyright statement.

Revert "Improving code"

This reverts commit 4bb030ab747fe710dcc43af68e501e6287ef8f0d.

Improving code

Reducing the amount of imports, converting functions into ES6 arrow function always, adding missing end of line, removing unnecessary border style.

Reducing the amount of imports, converting functions into ES6 arrow function always, adding missing end of line, removing unnecessary border style.

Assembling SideBar Component

Assembling SideBar Component by putting together its three child components: ProfilePic, NavLinks and Footer.

Adding EOFL

Adding EOFL

Adding EOFL

Adding EOFL

Improving code

Reducing the amount of imports, converting functions into ES6 arrow function always, adding missing end of line, removing unnecessary border style, converting if...else statement into ternary operator.

Correcting typo error

Correcting an uppercase letter that should be lowercase

Refactoring list

Refactoring list to be the return of a function, avoiding code repeat

Adding letter to Avatar

Adding one more letter between Avatar tags to use a better acronym

Refactoring list

Refactoring list to use less callback functions

Refactoring links

Refactoring links to be the return of a function, avoiding code repeat

Adding EOFL

Adding EOFL

Removing border style

Removing unnecessary border style.

Creation of style rules

Creation of style rules for sizes and colors

Adding style rules

Adding more thickness rules

Changing style rules

Changing style rules to use predefined rules

Adding semi-colon

Adding missing semi-colon

Changing style rules

Changing style rules to use predefined rules

Correcting typo error

Correcting wrong name of variable

Adding import

Adding missing import of style rules

Adding style rules

Adding thickness style rule

Changing style rules

Changing style rules to use predefined rules

Adding style rules

Adding thickness style rule

Changing style rules

Changing thickness style rule

Changing style rules

Changing style rules to use predefined rules

Adding style rules

Adding style rules and a new classification for thickness values, with a comment explaining why

Changing style rules

Changing style rules to use new value

Creation of IconMaker

Creation of a fourth component of sidebar, the IconMaker, which contains logic to return diferent icons based on received props.

Refactoring Footer

Refactoring the component Footer to use the new component IconMaker

Refactoring NavLinks

Refactoring the component NavLinks to use the new component IconMaker

Refactoring imports

Refactoring the Icon Maker component to use less imports

Refactoring imports

Refactoring the component to use less imports

Changing import

Importing without specifying /index

Moving export

Moving export to the left

Adding and editing style rules

Adding thickness rules and editing color rules

Changing style rules

Changing styles  to use new rules

Changing style rules

Changing styles  to use new rules

Changing style rules and passing props

Changing styles to use new rules and passing props to ProfilePic

Changing style rules and receiving props

Changing styles to use new rules and receiving props from SideBar.jsx to determine the picture displayed

Adding base project

Revert "Improving code"

This reverts commit 4bb030ab747fe710dcc43af68e501e6287ef8f0d.

Reducing the amount of imports, converting functions into ES6 arrow function always, adding missing end of line, removing unnecessary border style.

Adding EOFL

Adding EOFL

Adding EOFL

Adding EOFL

Correcting typo error

Correcting an uppercase letter that should be lowercase

Adding EOFL

Adding EOFL

Correcting typo error

Correcting wrong name of variable

Creation of IconMaker

Creation of a fourth component of sidebar, the IconMaker, which contains logic to return diferent icons based on received props.

Refactoring Footer

Refactoring the component Footer to use the new component IconMaker

Refactoring NavLinks

Refactoring the component NavLinks to use the new component IconMaker

Refactoring imports

Refactoring the component to use less imports

Changing import

Importing without specifying /index

Moving export

Moving export to the left

Adding and editing style rules

Adding thickness rules and editing color rules

Changing style rules

Changing styles  to use new rules

Changing style rules

Changing styles  to use new rules

Changing style rules and passing props

Changing styles to use new rules and passing props to ProfilePic

Changing style rules and receiving props

Changing styles to use new rules and receiving props from SideBar.jsx to determine the picture displayed

Adding dependencies

Adding @material-ui/core and @material-ui/icons dependencies to the project

Assembling SideBar Component

Assembling SideBar Component by putting together its three child components: ProfilePic, NavLinks and Footer.
Adding EOFL

Adding EOFL
Adding EOFL

Adding EOFL
Improving code

Reducing the amount of imports, converting functions into ES6 arrow function always, adding missing end of line, removing unnecessary border style, converting if...else statement into ternary operator.
Correcting typo error

Correcting an uppercase letter that should be lowercase
Refactoring list

Refactoring list to be the return of a function, avoiding code repeat
Adding letter to Avatar

Adding one more letter between Avatar tags to use a better acronym
Refactoring list

Refactoring list to use less callback functions
Refactoring links

Refactoring links to be the return of a function, avoiding code repeat
Adding EOFL

Adding EOFL
Removing border style

Removing unnecessary border style.
Creation of style rules

Creation of style rules for sizes and colors
Adding style rules

Adding more thickness rules
Changing style rules

Changing style rules to use predefined rules
Adding semi-colon

Adding missing semi-colon
Changing style rules

Changing style rules to use predefined rules
Correcting typo error

Correcting wrong name of variable
Adding import

Adding missing import of style rules
Adding style rules

Adding thickness style rule
Changing style rules

Changing style rules to use predefined rules
Adding style rules

Adding thickness style rule
Changing style rules

Changing thickness style rule
Changing style rules

Changing style rules to use predefined rules
Adding style rules

Adding style rules and a new classification for thickness values, with a comment explaining why
Changing style rules

Changing style rules to use new value
Creation of IconMaker

Creation of a fourth component of sidebar, the IconMaker, which contains logic to return diferent icons based on received props.
Refactoring Footer

Refactoring the component Footer to use the new component IconMaker
Refactoring NavLinks

Refactoring the component NavLinks to use the new component IconMaker
Refactoring imports

Refactoring the component to use less imports
Changing import

Importing without specifying /index
Moving export

Moving export to the left
Adding and editing style rules

Adding thickness rules and editing color rules
Changing style rules

Changing styles  to use new rules
Changing style rules

Changing styles  to use new rules
Changing style rules and passing props

Changing styles to use new rules and passing props to ProfilePic
Changing style rules and receiving props

Changing styles to use new rules and receiving props from SideBar.jsx to determine the picture displayed

Moving generic components

Moving generic components to root level

Creation of index at component level

Creation of index at component level to import components and then export them, simplifying the imports in SideBar.jsx

Simplifying imports

Simplifying imports using new index.js

Changing name of folder

Changing name of folder, removing the ".js" at the end

Changing import

Change import of IconMaker, due to the new name of its main folder

Receiving props

Receiving props correctly

Refactoring imports

Refactoring named imports to be default individual imports, because the @material-ui/icons dependency does not allow named imports for that components

Correcting import

Correcting the relative path of an import

Editing render method

Editing render method to render SideBar component

Changing styles

Changing styles to be more responsive

Changing border style

Changing border style to be on one side only

Removing "/"

Removing "/" from import's relative path

Correcting typo error

Correcting typo error in the URL of a link

Improving security

Improving security of links

Creation of ProfilePic

Creation of the first component of sidebar, the ProfilePic, which contains a profile picture and a name

Creation of NavLinks

Creation of the second component of sidebar, the NavLinks, which contains a list of navigation links

samuelouteda committed:
Creation of NavLinks

Creation of the second component of sidebar, the NavLinks, which contains a list of navigation links

samuelouteda committed:
Creation of NavLinks

Creation of the second component of sidebar, the NavLinks, which contains a list of navigation links

kilordpepo commented on src/routes.js:
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { BrowserRouter, Route, Switch, Redirect } from 'react-router-dom';
+import { SideBar } from './sections/SideBar/';
+import { Home } from './sections/Home/';
+import { AboutMe } from './sections/AboutMe/';
+import { Resume } from './sections/Resume/';
+import { ContactMe } from './sections/ContactMe/';
+import { NotFound } from './sections/NotFound/';
```
Please create an index file at /section level and then export from that file all those sections so that you can then do the following:

`import { NotFound, SideBar, Home , ...  } from './sections';`

kilordpepo commented on src/routes.js:
```diff
@@ -0,0 +1,38 @@
+      </div>
+    </BrowserRouter>
+  );
+};
```
add missing end of line here ( just one line more below that's empty)

kilordpepo commented on src/sections/SideBar/components/NavLinks/SideBarLinks.js:
```diff
@@ -0,0 +1,120 @@
+import React from "react";
+import { makeStyles } from "@material-ui/core/styles";
+import List from "@material-ui/core/List";
+import ListItem from "@material-ui/core/ListItem";
+import ListItemIcon from "@material-ui/core/ListItemIcon";
+import ListItemText from "@material-ui/core/ListItemText";
+import HomeOutlinedIcon from "@material-ui/icons/HomeOutlined";
+import PersonOutlineOutlinedIcon from "@material-ui/icons/PersonOutlineOutlined";
+import MailOutlineOutlinedIcon from "@material-ui/icons/MailOutlineOutlined";
+import NotesOutlinedIcon from "@material-ui/icons/NotesOutlined";
+import { createMuiTheme, ThemeProvider } from "@material-ui/core/styles";
+import Link from "@material-ui/core/Link";
```
Verify it's possible to do the same here with material ui, so that we can reduce the amount of imports

samuelouteda replied:
Something like this?:

```suggestion
import { makeStyles, List, ListItem, ListItemIcon, ... } from "@material-ui/core";
import { HomeOutlinedIcon, PersonOutlineOutlinedIcon, ... } from "@material-ui/icons";
```

kilordpepo replied:
exactly

kilordpepo commented on src/sections/SideBar/components/NavLinks/SideBarLinks.js:
```diff
@@ -0,0 +1,120 @@
+    maxWidth: 200,
+    backgroundColor: "rgb(61, 61, 61)",
+    color: "gray",
+    border: "1px solid black",
```
Let's investigate and add a style rule so that we can define rules for sizes on px with material, for example: 


    border: `${theme.size.1} solid black`, 
or something alike

kilordpepo replied:
The idea is to avoid having random sizes if not a defined o prefixed size for all the components

kilordpepo commented on src/sections/SideBar/components/NavLinks/SideBarLinks.js:
```diff
@@ -0,0 +1,120 @@
+  }
+});
+
+export default function SelectedListItem() {
```
convert to ES6 arrow function aways

kilordpepo commented on src/sections/SideBar/components/NavLinks/SideBarLinks.js:
```diff
@@ -0,0 +1,120 @@
+  };
+
+  const selectIconColor = (index) => {
+    if (selectedIndex === index) {
```
convert to ternary operator

kilordpepo commented on src/sections/SideBar/components/NavLinks/SideBarLinks.js:
```diff
@@ -0,0 +1,120 @@
+          component="nav"
+          aria-label="main mailbox folders"
+        >
+          <Link underline="none" color="inherit">
+            <ListItem
+              button
+              selected={selectedIndex === 0}
+              onClick={(event) => handleListItemClick(event, 0)}
+            >
+              <ListItemIcon>
+                <HomeOutlinedIcon className={selectIconColor(0)} />
+              </ListItemIcon>
+              <ListItemText primary="Home" />
+            </ListItem>
+          </Link>
```
It seem's this is a pattern is there any way you could create a function/component that returns this and use it here to avoid code repeat?

kilordpepo commented on src/sections/SideBar/components/NavLinks/SideBarLinks.js:
```diff
@@ -0,0 +1,120 @@
+      </ThemeProvider>
+    </div>
+  );
+}
```
add missing EOFL

kilordpepo commented on src/sections/SideBar/components/ProfilePic/SideBarProfile.js:
```diff
@@ -0,0 +1,49 @@
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    width: "16%",
+    minWidth: 200,
+    maxWidth: 240,
```
as before, let's see if we can define a rule for sizes

kilordpepo commented on src/sections/SideBar/components/ProfilePic/SideBarProfile.js:
```diff
@@ -0,0 +1,49 @@
+  }
+}));
+
+export default function ProfilePic() {
```
convert to ES6 arrow function

kilordpepo commented on src/sections/SideBar/components/ProfilePic/SideBarProfile.js:
```diff
@@ -0,0 +1,49 @@
+      className={classes.pic} 
+      src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAkGBxISEhITEhAWFRIVFRcVFRUVFRUVEBUQFRUWFhUWFRYYHSggGBolGxUVITEhJSkrLi4uFx8zODMtNygtLisBCgoKDg0OGhAQGi0lHx0tLS0rLSstLS0rLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0rLS0tLS0tLS0tLS0rLf/AABEIAOwA1gMBEQACEQEDEQH/xAAcAAEAAgMBAQEAAAAAAAAAAAAABAUCAwYBBwj/xAA8EAACAQIEBAQEBAMHBQEAAAAAAQIDEQQFITESQVFxBmGBsSIykaEHE1LBQtHhFCNicoKS8BVDosLxNP/EABoBAQADA ...
+      >
+        C
```
typo error?

samuelouteda replied:
It's a feature of Material UI. If the Avatar element cannot load the image, it displays the text between its tags instead.

kilordpepo replied:
Then put my name just in case

samuelouteda replied:
Got it, sir!

kilordpepo commented on src/sections/SideBar/components/ProfilePic/SideBarProfile.js:
```diff
@@ -0,0 +1,49 @@
+      </Typography>
+    </div>
+  )
+}
```
add end of line here

kilordpepo reviewed (commented):
Please investigate and fix these issues and attach an image on the PR description along with the sections covered in this PR

samuelouteda committed:
Creating file to import all sections

Creating file to import all sections and export where needed

samuelouteda committed:
Creating file to import all sections

Creating file to import all sections and export where needed

samuelouteda committed:
Simplifying imports

Using exports from index.js to simplify imports

samuelouteda committed:
Simplifying imports

Using exports from index.js to simplify imports

samuelouteda committed:
Improving code

Reducing the amount of imports, converting functions into ES6 arrow function aways, adding missing end of line

samuelouteda committed:
Revert "Improving code"

This reverts commit 4bb030ab747fe710dcc43af68e501e6287ef8f0d.

samuelouteda committed:
Revert "Improving code"

This reverts commit 4bb030ab747fe710dcc43af68e501e6287ef8f0d.

samuelouteda committed:
Improving code

Reducing the amount of imports, converting functions into ES6 arrow function aways, adding missing end of line

samuelouteda committed:
Creation of Footer

Creation of the third and last component of sidebar, the Footer, which contains links to social networks with their icons and copyright statement.

samuelouteda committed:
Creation of Footer

Creation of the third and last component of sidebar, the Footer, which contains links to social networks with their icons and copyright statement.

samuelouteda committed:
Reducing the amount of imports, converting functions into ES6 arrow function always, adding missing end of line, removing unnecessary border style.

samuelouteda committed:
Creation of routes

Creation of ProfilePic

Creation of the first component of sidebar, the ProfilePic, which contains a profile picture and a name
Creation of NavLinks

Creation of the second component of sidebar, the NavLinks, which contains a list of navigation links
Creating file to import all sections

Creating file to import all sections and export where needed
Simplifying imports

Using exports from index.js to simplify imports
Improving code

Reducing the amount of imports, converting functions into ES6 arrow function aways, adding missing end of line
Creation of Footer

Creation of the third and last component of sidebar, the Footer, which contains links to social networks with their icons and copyright statement.
Revert "Improving code"

This reverts commit 4bb030ab747fe710dcc43af68e501e6287ef8f0d.

Reducing the amount of imports, converting functions into ES6 arrow function always, adding missing end of line, removing unnecessary border style.

samuelouteda committed:
Reducing the amount of imports, converting functions into ES6 arrow function always, adding missing end of line, removing unnecessary border style.

samuelouteda committed:
Improving code

Reducing the amount of imports, converting functions into ES6 arrow function always, adding missing end of line, removing unnecessary border style.

samuelouteda committed:
Merge remote-tracking branch 'origin/develop' into develop

kilordpepo reviewed (commented):
Careful with git, I saw a revert command you did, I would suggest you read about git reset --soft instead of using revert and after do a force push

kilordpepo commented on src/sections/SideBar/components/ProfilePic/SideBarProfile.js:
```diff
@@ -45,5 +42,7 @@ export default function ProfilePic() {
+  );
+};
+
+export default ProfilePic;
```
Add EOFL

kilordpepo commented on src/sections/index.js:
```diff
@@ -0,0 +1,8 @@
+import Resume from "./Resume";
+import SideBar from "./SideBar";
+
+export { AboutMe, ContactMe, Home, NotFound, Resume, SideBar };
```
EOFL here

kilordpepo reviewed (commented):
I also suggest you download and use two pluggins you can add to VScode, one is GitLens, the other is ESLint with prettier

The first will help you seeing more details of git in vscode, the last 2 will lint the code as you save so it's better looking and easier to read

samuelouteda committed:
Assembling SideBar Component

Assembling SideBar Component by putting together its three child components: ProfilePic, NavLinks and Footer.

samuelouteda committed:
Assembling SideBar Component

Assembling SideBar Component by putting together its three child components: ProfilePic, NavLinks and Footer.

samuelouteda committed:
Adding EOFL

Adding EOFL

samuelouteda committed:
Adding EOFL

Adding EOFL

samuelouteda committed:
Adding EOFL

Adding EOFL

samuelouteda committed:
Adding EOFL

Adding EOFL

samuelouteda committed:
Improving code

Reducing the amount of imports, converting functions into ES6 arrow function always, adding missing end of line, removing unnecessary border style, converting if...else statement into ternary operator.

samuelouteda committed:
Improving code

Reducing the amount of imports, converting functions into ES6 arrow function always, adding missing end of line, removing unnecessary border style, converting if...else statement into ternary operator.

SamuelOuteda commented:
Assembled SideBar Component
![image](https://user-images.githubusercontent.com/74317636/99733845-30612300-2a98-11eb-9644-2b33d3fb5d20.png)

ProfilePic child component
![image](https://user-images.githubusercontent.com/74317636/99734536-563af780-2a99-11eb-9549-03a00e3bffb4.png)

NavLinks child component
![image](https://user-images.githubusercontent.com/74317636/99734673-95694880-2a99-11eb-83d0-30b2fe968d05.png)

Footer child component
![image](https://user-images.githubusercontent.com/74317636/99734704-a87c1880-2a99-11eb-9312-e762b7075977.png)

samuelouteda committed:
Correcting typo error

Correcting an uppercase letter that should be lowercase

samuelouteda committed:
Correcting typo error

Correcting an uppercase letter that should be lowercase

samuelouteda committed:
Refactoring list

Refactoring list to be the return of a function, avoiding code repeat

samuelouteda committed:
Refactoring list

Refactoring list to be the return of a function, avoiding code repeat

samuelouteda committed:
Adding letter to Avatar

Adding one more letter between Avatar tags to use a better acronym

samuelouteda committed:
Adding letter to Avatar

Adding one more letter between Avatar tags to use a better acronym

samuelouteda committed:
Refactoring list

Refactoring list to use less callback functions

samuelouteda committed:
Refactoring list

Refactoring list to use less callback functions

samuelouteda committed:
Refactoring links

Refactoring links to be the return of a function, avoiding code repeat

samuelouteda committed:
Refactoring links

Refactoring links to be the return of a function, avoiding code repeat

kilordpepo commented on src/sections/SideBar/SideBar.jsx:
```diff
@@ -0,0 +1,36 @@
+  )
+}
+
+export default SideBar;
```
missing EOF

kilordpepo commented on src/sections/SideBar/components/Footer/SideBarFooter.js:
```diff
@@ -0,0 +1,92 @@
+  onMouseOver={(event) => handleMouseOver(event, 1)} 
+  onMouseOut={(event) => handleMouseOver(event, 0)} 
+  />], 
+  ["https://www.instagram.com/", <InstagramIcon 
```
Seems like a component, walks like a component and quacks like one then let's make this an Icon component! that accepts an argument string to specify which icon is going to render and also receives both events. By doing this you should be able to avoid setting the component inside the array and just rendering that generic icon component

kilordpepo commented on src/sections/SideBar/components/Footer/SideBarFooter.js:
```diff
@@ -0,0 +1,92 @@
+      </div>
+      <div className={classes.container}>
+        <Typography className={classes.text} variant="body2">
+          2020 &copy; Paola & Pepo, C.A.<br />All Rights Reserved.
```
🎈

kilordpepo commented on src/sections/SideBar/components/NavLinks/SideBarLinks.js:
```diff
@@ -0,0 +1,80 @@
+    return (selectedIndex === index) ? classes.icon2 : classes.icon;
+  };
+
+  const arrValues = [["Home", <HomeOutlinedIcon className={selectIconColor(0)} />, "./home"], 
```
the same generic component I mentioned before should help us here too, avoid setting up components inside a variable whenever is possible

samuelouteda committed:
Adding EOFL

Adding EOFL

samuelouteda committed:
Adding EOFL

Adding EOFL

kilordpepo commented on src/sections/SideBar/components/Footer/SideBarFooter.js:
```diff
@@ -0,0 +1,92 @@
+    minWidth: 200,
+    maxWidth: 240,
+    backgroundColor: "rgb(61, 61, 61)",
+    border: "1px solid black",
```
Remove

samuelouteda committed:
Removing border style

Removing unnecessary border style.

samuelouteda committed:
Removing border style

Removing unnecessary border style.

samuelouteda committed:
Creation of style rules

Creation of style rules for sizes and colors

samuelouteda committed:
Creation of style rules

Creation of style rules for sizes and colors

samuelouteda committed:
Adding style rules

Adding more thickness rules

samuelouteda committed:
Adding style rules

Adding more thickness rules

samuelouteda committed:
Changing style rules

Changing style rules to use predefined rules

samuelouteda committed:
Changing style rules

Changing style rules to use predefined rules

samuelouteda committed:
Adding semi-colon

Adding missing semi-colon

samuelouteda committed:
Adding semi-colon

Adding missing semi-colon

samuelouteda committed:
Changing style rules

Changing style rules to use predefined rules

samuelouteda committed:
Changing style rules

Changing style rules to use predefined rules

samuelouteda committed:
Correcting typo error

Correcting wrong name of variable

samuelouteda committed:
Correcting typo error

Correcting wrong name of variable

samuelouteda committed:
Adding import

Adding missing import of style rules

samuelouteda committed:
Adding import

Adding missing import of style rules

samuelouteda committed:
Adding style rules

Adding thickness style rule

samuelouteda committed:
Adding style rules

Adding thickness style rule

samuelouteda committed:
Changing style rules

Changing style rules to use predefined rules

samuelouteda committed:
Changing style rules

Changing style rules to use predefined rules

samuelouteda committed:
Adding style rules

Adding thickness style rule

samuelouteda committed:
Adding style rules

Adding thickness style rule

samuelouteda committed:
Changing style rules

Changing thickness style rule

samuelouteda committed:
Changing style rules

Changing thickness style rule

samuelouteda committed:
Changing style rules

Changing style rules to use predefined rules

samuelouteda committed:
Changing style rules

Changing style rules to use predefined rules

samuelouteda committed:
Adding style rules

Adding style rules and a new classification for thickness values, with a comment explaining why

samuelouteda committed:
Adding style rules

Adding style rules and a new classification for thickness values, with a comment explaining why

samuelouteda committed:
Changing style rules

Changing style rules to use new value

samuelouteda committed:
Changing style rules

Changing style rules to use new value

samuelouteda committed:
Creation of IconMaker

Creation of a fourth component of sidebar, the IconMaker, which contains logic to return diferent icons based on received props.

samuelouteda committed:
Creation of IconMaker

Creation of a fourth component of sidebar, the IconMaker, which contains logic to return diferent icons based on received props.

samuelouteda committed:
Refactoring Footer

Refactoring the component Footer to use the new component IconMaker

samuelouteda committed:
Refactoring Footer

Refactoring the component Footer to use the new component IconMaker

samuelouteda committed:
Refactoring NavLinks

Refactoring the component NavLinks to use the new component IconMaker

samuelouteda committed:
Refactoring NavLinks

Refactoring the component NavLinks to use the new component IconMaker

samuelouteda committed:
Refactoring imports

Refactoring the component to use less imports

samuelouteda committed:
Refactoring imports

Refactoring the component to use less imports

samuelouteda committed:
Refactoring imports

Refactoring the Icon Maker component to use less imports

samuelouteda committed:
Merge remote-tracking branch 'origin/develop' into develop

kilordpepo commented on src/routes.js:
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { BrowserRouter, Route, Switch, Redirect } from 'react-router-dom';
+import { SideBar, Home, AboutMe, Resume, ContactMe, NotFound } from './sections/index';
```
You should be able to import this from sections directly without specifying /index

kilordpepo commented on src/sections/SideBar/SideBar.jsx:
```diff
@@ -0,0 +1,37 @@
+import Footer from "./components/Footer/SideBarFooter";
+import SelectedListItem from "./components/NavLinks/SideBarLinks";
+import ProfilePic from "./components/ProfilePic/SideBarProfile";
+import { styles } from "../../../../styles/styles";
```
Please investigate how to define relative routes so that we can start importing files like from `styles` and `components/ProfilePic` also please create an index file on components level that will export all the contained components so that you can import it from `./components`. Also **All generic components** like SideBarProfile, or SideBarLink should be located on a folder named `components` at root level. they can be used or shared between different components so don't define it inside the components of a section, it should be a generic component you will be using in different files

kilordpepo commented on src/sections/SideBar/components/Footer/SideBarFooter.js:
```diff
@@ -0,0 +1,82 @@
+import React from "react";
```
Here too, please move the footer component outside to the root level so that we can use it in other components just in case

kilordpepo commented on src/sections/SideBar/components/IconMaker.js/IconMaker.js:
```diff
@@ -0,0 +1,23 @@
+  };
+  
+  export default IconMaker;
+  
```
Investigate why it's telling you that you have EOF

kilordpepo commented on src/sections/SideBar/components/IconMaker.js/IconMaker.js:
```diff
@@ -0,0 +1,23 @@
+    );
+  };
+  
+  export default IconMaker;
```
move this to the left

kilordpepo commented on src/sections/SideBar/components/IconMaker.js/IconMaker.js:
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import { HomeOutlinedIcon, PersonOutlineOutlinedIcon, MailOutlineOutlinedIcon, NotesOutlinedIcon, FacebookIcon, InstagramIcon, LinkedInIcon, GitHubIcon } from "@material-ui/icons";
```
Use prettier or a linter plugin that will sort this in different lines

kilordpepo commented on src/sections/SideBar/components/NavLinks/SideBarLinks.js:
```diff
@@ -0,0 +1,82 @@
+import React from "react";
+import { makeStyles, Link, List, ListItem, ListItemIcon, ListItemText, createMuiTheme, ThemeProvider } from "@material-ui/core";
+import { styles } from "../../../../styles/styles";
```
same thing with the routes here

kilordpepo commented on src/sections/SideBar/components/NavLinks/SideBarLinks.js:
```diff
@@ -0,0 +1,82 @@
+import React from "react";
```
Move this file to root components

kilordpepo commented on src/sections/SideBar/components/ProfilePic/SideBarProfile.js:
```diff
@@ -0,0 +1,49 @@
+  }
+}));
+
+const ProfilePic = () => {
```
This seems to be a generic profile pic we could use in other sections, please move to root level

kilordpepo replied:
Please add props such as src, alt, .... so that we can specify another image or alt to the Avatar

kilordpepo commented on src/styles/styles.js:
```diff
@@ -0,0 +1,34 @@
+export const styles = {
+  colors: {
+    gray1: "rgb(45, 45, 45)",
```
more specified colors would be better so that it make sense such as lightGray, darkGray, limeGray and so on, you can't know the colors scale with just a number. I prefer the variable name to be longer and descriptive thant generic and non readable, **this applies to all variable names** in the code

kilordpepo commented on src/styles/styles.js:
```diff
@@ -0,0 +1,34 @@
+      thick2: 2,
+      thick3: 4,
+      thick4: 5,
+      thick5: 10,
```
I think jumping from 5 to 5 will be problematic. Let's jump 2 from to and define 20 thicks

kilordpepo commented on src/styles/styles.js:
```diff
@@ -0,0 +1,34 @@
+      thick13: 50
+    },
+    specialThick: { //This rule is for really specific numbers. This way, that values do not push the normal ones.
+      thick1: 42
```
since this is a specialThick can you define a special name for it? so that we can know why or where it's used

kilordpepo reviewed (commented):
Let's keep making this changes, the idea is that you can abstract from this what's a component and what's a section component and make the code more readable. Please **use** prettier or some linter and apply it to all files. Also please `rebase` so that we can reduce the amount of commits, the more commits you do, then the more the work is going to be

samuelouteda committed:
Changing import

Importing without specifying /index

samuelouteda committed:
Changing import

Importing without specifying /index

samuelouteda committed:
Moving export

Moving export to the left

samuelouteda committed:
Moving export

Moving export to the left

samuelouteda committed:
Adding and editing style rules

Adding thickness rules and editing color rules

samuelouteda committed:
Adding and editing style rules

Adding thickness rules and editing color rules

samuelouteda committed:
Changing style rules

Changing styles  to use new rules

samuelouteda committed:
Changing style rules

Changing styles  to use new rules

samuelouteda committed:
Changing style rules

Changing styles  to use new rules

samuelouteda committed:
Changing style rules

Changing styles  to use new rules

samuelouteda committed:
Changing style rules and passing props

Changing styles to use new rules and passing props to ProfilePic

samuelouteda committed:
Changing style rules and passing props

Changing styles to use new rules and passing props to ProfilePic

samuelouteda committed:
Changing style rules and receiving props

Changing styles to use new rules and receiving props from SideBar.jsx to determine the picture displayed

samuelouteda committed:
Assembling SideBar Component

Assembling SideBar Component by putting together its three child components: ProfilePic, NavLinks and Footer.
Adding EOFL

Adding EOFL
Adding EOFL

Adding EOFL
Improving code

Reducing the amount of imports, converting functions into ES6 arrow function always, adding missing end of line, removing unnecessary border style, converting if...else statement into ternary operator.
Correcting typo error

Correcting an uppercase letter that should be lowercase
Refactoring list

Refactoring list to be the return of a function, avoiding code repeat
Adding letter to Avatar

Adding one more letter between Avatar tags to use a better acronym
Refactoring list

Refactoring list to use less callback functions
Refactoring links

Refactoring links to be the return of a function, avoiding code repeat
Adding EOFL

Adding EOFL
Removing border style

Removing unnecessary border style.
Creation of style rules

Creation of style rules for sizes and colors
Adding style rules

Adding more thickness rules
Changing style rules

Changing style rules to use predefined rules
Adding semi-colon

Adding missing semi-colon
Changing style rules

Changing style rules to use predefined rules
Correcting typo error

Correcting wrong name of variable
Adding import

Adding missing import of style rules
Adding style rules

Adding thickness style rule
Changing style rules

Changing style rules to use predefined rules
Adding style rules

Adding thickness style rule
Changing style rules

Changing thickness style rule
Changing style rules

Changing style rules to use predefined rules
Adding style rules

Adding style rules and a new classification for thickness values, with a comment explaining why
Changing style rules

Changing style rules to use new value
Creation of IconMaker

Creation of a fourth component of sidebar, the IconMaker, which contains logic to return diferent icons based on received props.
Refactoring Footer

Refactoring the component Footer to use the new component IconMaker
Refactoring NavLinks

Refactoring the component NavLinks to use the new component IconMaker
Refactoring imports

Refactoring the component to use less imports
Changing import

Importing without specifying /index
Moving export

Moving export to the left
Adding and editing style rules

Adding thickness rules and editing color rules
Changing style rules

Changing styles  to use new rules
Changing style rules

Changing styles  to use new rules
Changing style rules and passing props

Changing styles to use new rules and passing props to ProfilePic
Changing style rules and receiving props

Changing styles to use new rules and receiving props from SideBar.jsx to determine the picture displayed

samuelouteda committed:
Changing style rules and receiving props

Changing styles to use new rules and receiving props from SideBar.jsx to determine the picture displayed

kilordpepo commented on src/routes.js:
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { BrowserRouter, Route, Switch, Redirect } from 'react-router-dom';
+import { SideBar, Home, AboutMe, Resume, ContactMe, NotFound } from './sections/';
```
you can remove` /`

samuelouteda committed:
Merge remote-tracking branch 'origin/develop' into develop

samuelouteda committed:
Adding dependencies

Adding @material-ui/core and @material-ui/icons dependencies to the project

samuelouteda committed:
Merge remote-tracking branch 'origin/develop' into develop

samuelouteda committed:
Moving generic components

Moving generic components to root level

samuelouteda committed:
Creation of index at component level

Creation of index at component level to import components and then export them, simplifying the imports in SideBar.jsx

samuelouteda committed:
Simplifying imports

Simplifying imports using new index.js

samuelouteda committed:
Changing name of folder

Changing name of folder, removing the ".js" at the end

samuelouteda committed:
Changing import

Change import of IconMaker, due to the new name of its main folder

samuelouteda committed:
Receiving props

Receiving props correctly

samuelouteda committed:
Refactoring imports

Refactoring named imports to be default individual imports, because the @material-ui/icons dependency does not allow named imports for that components

samuelouteda committed:
Correcting import

Correcting the relative path of an import

samuelouteda committed:
Editing render method

Editing render method to render SideBar component

samuelouteda committed:
Changing styles

Changing styles to be more responsive

samuelouteda committed:
Changing border style

Changing border style to be on one side only

samuelouteda committed:
Removing "/"

Removing "/" from import's relative path

samuelouteda committed:
Correcting typo error

Correcting typo error in the URL of a link

samuelouteda committed:
Improving security

Improving security of links

kilordpepo commented:
Please rebase you changes as there's about 112 commits

kilordpepo commented on src/App.js:
```diff
@@ -1,23 +1,9 @@
-import './App.css';
+import SideBar from "./sections/SideBar/SideBar";
 
 function App() {
```
convert to ES6 function

kilordpepo commented on src/components/NavLinks/SideBarLinks.js:
```diff
@@ -0,0 +1,82 @@
+import React from "react";
+import { makeStyles, Link, List, ListItem, ListItemIcon, ListItemText, createMuiTheme, ThemeProvider } from "@material-ui/core";
+import { styles } from "../../styles/styles";
+import IconMaker from '../../sections/SideBar/components/IconMaker/IconMaker';
```
If this is a component for all the aplications there shouldn't be anything imported from sections, in any case this is another component or a subcomponent of SideBarLinks

kilordpepo commented on src/components/NavLinks/SideBarLinks.js:
```diff
@@ -0,0 +1,82 @@
+    return (selectedIndex === index) ? classes.icon2 : classes.icon;
+  };
+
+  const arrValues = [["Home", "./home"], 
```
I think this should be passed as props

kilordpepo commented on src/components/ProfilePic/SideBarProfile.js:
```diff
@@ -0,0 +1,50 @@
+  }
+}));
+
+const ProfilePic = (props) => {
```
Please destructure the props and use the ones neded, never receive props without destructuring unless necessary

kilordpepo commented on src/sections/SideBar/SideBar.jsx:
```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import { makeStyles } from "@material-ui/core/styles";
+import { Footer, SelectedListItem, ProfilePic } from "./components/index";
+import { styles } from "../../styles/styles";
```
see if we can short the links like importing as `styles` or something.

kilordpepo commented on src/sections/SideBar/SideBar.jsx:
```diff
@@ -0,0 +1,43 @@
+    maxWidth: styles.sizes.sidebarDimensions.maxWidth,
+    height: "100%",
+    backgroundColor: styles.colors.darkGray,
+    borderRight: `${styles.sizes.thickness.thick1}px solid black`
```
can you make thicks have the `px `on it so that you don't have to add it everytime?

kilordpepo commented on src/sections/SideBar/components/Footer/SideBarFooter.js:
```diff
@@ -0,0 +1,83 @@
+import React from "react";
+import { makeStyles, Link, Typography } from "@material-ui/core";
+import { styles } from "../../../../styles/styles";
```
same here with the imports, see if we can decrease the amount of paths to /styles by adding some rule somewhere that knows importing from /styles means that route

kilordpepo commented on src/sections/SideBar/components/Footer/SideBarFooter.js:
```diff
@@ -0,0 +1,83 @@
+import React from "react";
+import { makeStyles, Link, Typography } from "@material-ui/core";
+import { styles } from "../../../../styles/styles";
+import IconMaker from '../IconMaker/IconMaker';
```
add an index so this is not redudant

kilordpepo commented on src/sections/SideBar/components/Footer/SideBarFooter.js:
```diff
@@ -0,0 +1,83 @@
+    return (selectedIndex === index) ? classes.icon2 : classes.icon;
+  };
+
+  const arrValues = ["https://www.facebook.com/", 
```
should be received as props

kilordpepo commented on src/sections/SideBar/components/IconMaker/IconMaker.js:
```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import HomeOutlinedIcon from "@material-ui/icons/HomeOutlined";
```
any way we can import all the icons just from one or two routes?

kilordpepo commented on src/sections/SideBar/components/IconMaker/IconMaker.js:
```diff
@@ -0,0 +1,30 @@
+
+const IconMaker = (props) => {
+    const selectIcon = (index) => {
+      return index === 0 ? <HomeOutlinedIcon className={props.icon} />
```
I prefer insted of use a numer use something else more descriptive as if `icons === "home" ` also define those icon names in a constant object where you can use it everywhere else. Please don't use ternary operators conncatenated like this as is a bad practice. Instead create an Iconst object that contains properties you can access with the icon, for example:

kilordpepo replied:
```suggestion
constants file:

export const ICONS = {
home: "home",
person: "person",
....}

IconMaker:

const ICON_ELEMENTS = {
[ICONS.home] : <HomeOutlined.. >,
[ICONS.person]: <PersonOutlined...>
}

return ICON_ELEMENTS[receivedIcon]


```

kilordpepo replied:
But I rather prefer conditional rendering so please do something like:
```suggestion
      return 
      <> 
      {receivedIcon === ICONS.home && <HomeIcon>}
      {receivedIcon === ICONS.person && <PersonIcon>}
      </>
```

kilordpepo commented on src/sections/SideBar/components/IconMaker/IconMaker.js:
```diff
@@ -0,0 +1,30 @@
+  };
+  
+export default IconMaker;
+  
```
missing EOF

kilordpepo commented on src/sections/SideBar/components/index.js:
```diff
@@ -0,0 +1,5 @@
+import Footer from './Footer/SideBarFooter';
+import SelectedListItem from '../../../components/NavLinks/SideBarLinks';
```
Same comments here about routes

kilordpepo reviewed (commented):
Some minor comments

samuelouteda committed:
Merge remote-tracking branch 'origin/develop' into develop

samuelouteda committed:
Improving code

Removing legacy files, refactoring App's component to ES6 arrow function syntax, adding 'px' unit to style rules, exporting IconMaker in components/index to import it in other files without being redundant, importing SideBar in App.js from sections/index, adding component's empty files to import them on sections/index (because it needs to import something, otherwise it throws an error), passing props to SideBar.jsx, receiving props on SideBarLinks and SideBarFooter, passing only the necessary props through destructuring, changing style rule on SideBarFooter to use predefined rules, adding new index on IconMaker's folder to export all icons and then import them on IconMaker.js in one line, adding propTypes to SideBarLinks, SideBarFooter, SideBarProfile and IconMaker, refactoring IconMaker to use if...else if statements with descriptive received props to select the icon instead of concatenated ternary operators with index numbers

samuelouteda committed:
Correcting propTypes

Correcting propTypes to use new props received

kilordpepo commented on src/components/NavLinks/SideBarLinks.js:
```diff
@@ -0,0 +1,84 @@
+  const classes = useStyles();
+  const [selectedIndex, setSelectedIndex] = React.useState(0);
+
+  const handleListItemClick = (event, index) => {
```
indeed use of {}, you can return the ternary operator without `{}`

kilordpepo commented on src/components/NavLinks/SideBarLinks.js:
```diff
@@ -0,0 +1,84 @@
+  };
+
+  const selectIconColor = (index) => {
+    return (selectedIndex === index) ? classes.icon2 : classes.icon;
```
indeed use of {}, you can return the ternary operator without `{}`

kilordpepo commented on src/components/NavLinks/SideBarLinks.js:
```diff
@@ -0,0 +1,84 @@
+    return (selectedIndex === index) ? classes.icon2 : classes.icon;
+  };
+
+  const arrValues = items;
```
remove arrValues and use only items prop

kilordpepo commented on src/components/NavLinks/SideBarLinks.js:
```diff
@@ -0,0 +1,84 @@
+  };
+
+SelectedListItem.propTypes = {
+  items: PropTypes.array.isRequired
```
use arrayOf and define the objects ,the idea is not to leave something unclear

kilordpepo commented on src/sections/SideBar/SideBar.jsx:
```diff
@@ -0,0 +1,49 @@
+          <ProfilePic alt="Carlos Valero" src="https://i.pinimg.com/236x/18/81/2e/18812e006fca40297d3f8be353a57c18--pig-girl-pig-birthday.jpg" />
+        </div>
+        <div className={classes.overlap}>
+          <SelectedListItem items={[["Home", "./home", "home"], 
```
please define them in 2 separate variables rather than defining the array inside the items

kilordpepo commented on src/sections/SideBar/components/Footer/SideBarFooter.js:
```diff
@@ -0,0 +1,85 @@
+
+const Footer = ({ links }) => {
+  const classes = useStyles();
+  const [selectedIndex, setSelectedIndex] = React.useState(0);
```
import useState directly from react don't do React.useState. For this do the following
```suggestion
import React,{useState} from "react";
```

kilordpepo commented on src/sections/SideBar/components/Footer/SideBarFooter.js:
```diff
@@ -0,0 +1,85 @@
+    setSelectedIndex(index);
+  };
+
+  const selectIconColor = (index) => {
```
no need for {}

kilordpepo commented on src/sections/SideBar/components/Footer/SideBarFooter.js:
```diff
@@ -0,0 +1,85 @@
+};
+
+Footer.propTypes = {
+  links: PropTypes.array.isRequired
```
arrayOf

kilordpepo commented on src/sections/SideBar/components/IconMaker/IconMaker.js:
```diff
@@ -0,0 +1,46 @@
+  GitHubIcon, icons } from "./index";
+
+const IconMaker = ({ icon, iconColor, mouseOver, mouseOut }) => {
+  const selectIcon = (icon) => {
```
no if else conditional rendering just 
```

{condition && element}
{condition && element}
{condition && element}
{condition && element}

```

kilordpepo replied:
like

kilordpepo replied:
`{condition && <element1> </element1>}`

kilordpepo replied:
Also please define icons in a constants js file at root level so that it can be used by all sections or components. Please make it Mayus like ICONS

kilordpepo commented on src/sections/SideBar/components/IconMaker/IconMaker.js:
```diff
@@ -0,0 +1,46 @@
+
+IconMaker.propTypes = {
+  icon: PropTypes.string.isRequired,
+  iconColor: PropTypes.object.isRequired,
```
use shape and define props

kilordpepo commented on src/sections/SideBar/components/IconMaker/iconsObject.js:
```diff
@@ -0,0 +1,10 @@
+    instagram: "instagram",
+    linkedIn: "linkedIn",
+    gitHub: "gitHub"
+}
```
EOF

kilordpepo commented on src/sections/SideBar/components/IconMaker/index.js:
```diff
@@ -0,0 +1,18 @@
+    FacebookIcon, 
+    InstagramIcon, 
+    LinkedInIcon, 
+    GitHubIcon, icons };
```
EOF

kilordpepo reviewed (commented):
Please rebase as I still see 118 lines. Just keep in mind that whenever you have the text editor opened and you have to pick the commits in the rebase, after you finish you can press Ctrl + X to save and exit when it's on ubuntu. Keep in mind

samuelouteda committed:
Improving code

Removing unnecessary angle brackets, removing arrValues and only using items props, using arrayOf and objectOf to define the received props, separating items props into two different props, importing useState directly from React, refactoring if else conditional rendering on IconMaker to use only and operators, defining icons in uppercase in a constants.js file, adding missing EOF

samuelouteda committed:
Applying ESLint

Applying ESLint with Prettier to all files

kilordpepo commented on src/components/NavLinks/SideBarLinks.js:
```diff
@@ -78,7 +72,8 @@ const SelectedListItem = ({ items }) => {
 
 SelectedListItem.propTypes = {
-  items: PropTypes.array.isRequired
+  items: PropTypes.arrayOf(PropTypes.array),
```
I think this is not properly defined, if it's a matix then `PropTypes.arrayOf(PropTypes.arrayOf(Proptypes.shape({something})))`?

kilordpepo reviewed (approved):
Let's see if we can do changes required for routes, if not then we can change this later. For the rest I see some improvements but nothing we can't fix later 👍
